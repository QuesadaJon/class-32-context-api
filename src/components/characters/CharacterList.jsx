import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map((character) => (
    <li key={character._id}>
      <Character
        name={character.name}
        image={character.photoUrl}
      />
    </li>
  ));
  return <ul data-testid="monsters">{characterElements}</ul>;
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    photoUrl: PropTypes.string.isRequired
  })
  ).isRequired
};

export default CharacterList;
