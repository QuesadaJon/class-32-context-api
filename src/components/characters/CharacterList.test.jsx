import React from 'react';
import { render } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('displays an character', () => {
  it('renders one single character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList 
          characters={[]} 
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
