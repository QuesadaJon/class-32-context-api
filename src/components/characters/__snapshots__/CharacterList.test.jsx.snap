// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an character renders one single character 1`] = `
<DocumentFragment>
  <ul
    data-testid="monsters"
  />
</DocumentFragment>
`;
